### PR TITLE
[FEC-96] Migrate react-router redirects

### PR DESCRIPTION
**Pull request by CarlosCortizasCT**
#### Summary

Migrate react-router redirects

#### Description

As part of the preparations for the upcoming `react-router` v6 migration, there are some changes related to how client-side redirects are managed ([reference](https://reactrouter.com/en/main/upgrading/v5#remove-redirects-inside-switch)) that we can already apply to our codebase (supported in v5).

### CODE REVIEW

**Review by mario-priceless — APPROVED** [review]

**Review by tdeekens — APPROVED** [review]

**Review by kark — APPROVED** [review]
Thank you Carlos 🙌
Just a friendly reminder to add the changeset before merging.
Many thanks

**Review by Rhotimee — APPROVED** [review]
Nice!

**Comment by CarlosCortizasCT** [comment]
> Thank you Carlos 🙌 Just a friendly reminder to add the changeset before merging. Many thanks

Thanks for the reminder, @kark 🙇 

Included here: 146330f102809a1145968ee10dabf5c0b152160e

File: packages/application-components/src/components/detail-pages/tabular-detail-page/tabular-detail-page.spec.tsx [hunk 1]
```diff
@@ -19,9 +19,7 @@ jest.mock('@commercetools-uikit/utils', () => ({
 const Content = () => (
   <Spacings.Stack scale="m">
     <Switch>
-      <Route exact={true} path="/">
-        <Redirect to="/tab-one" />
-      </Route>
```
Comment (emmenko) [inline packages/application-components/src/components/detail-pages/tabular-detail-page/tabular-detail-page.spec.tsx:24]: Nit: this is technically correct.


The docs say 

> Normal `<Redirect>` elements that are not inside a `<Switch>` are ok to remain. 

> Remove any `<Redirect>` elements that are directly inside a `<Switch>`.

So only `<Switch><Redirect /></Switch>` should be migrated.

Using `render` or `children` is also equivalent, although if I remember back then they recommended to prefer `children`.

**Review by emmenko — APPROVED** [review]
Thanks

File: packages/application-components/src/components/detail-pages/tabular-detail-page/tabular-detail-page.spec.tsx (hunk 1 above)
Comment (CarlosCortizasCT) [reply packages/application-components/src/components/detail-pages/tabular-detail-page/tabular-detail-page.spec.tsx:24]: Render `children` is discouraged in v6 as we will need to use the `element` property of the `Route` component so I thought it was easier to follow the prop recommendation.

Comment (emmenko) [reply packages/application-components/src/components/detail-pages/tabular-detail-page/tabular-detail-page.spec.tsx:24]: Yes, that makes sense. Thanks!

**Pull request merged**